test/test_unidler.py

```diff
@@ -0,0 +1,253 @@
+from io import BytesIO
+from unittest.mock import MagicMock, call, patch
+
+import pytest
+
+import unidler
+from unidler import (
+    IDLED,
+    IDLED_AT,
+    INGRESS_CLASS,
+    RequestHandler,
+    UNIDLER,
+)
+
+
+HOSTNAME = 'test.host.name'
+
+
+@pytest.yield_fixture
+def client():
+    client = MagicMock()
+    with patch('unidler.client', client):
+        yield client
+
+
+@pytest.fixture
+def deployment():
+    deployment = MagicMock()
+    deployment.spec.replicas = 0
+    deployment.metadata.labels = {
+        IDLED: 'true',
+    }
+    deployment.metadata.annotations = {
+        IDLED_AT: 'YYYY-mm-ddTHH:MM:SS+0000,2',
+    }
+    deployment.metadata.name = 'test-app'
+    deployment.metadata.namespace = 'test-namespace'
+    return deployment
+
+
+@pytest.fixture
+def ingress():
+    host_rule = MagicMock()
+    host_rule.host = HOSTNAME
+    ingress = MagicMock()
+    ingress.spec.rules = [
+        host_rule,
+    ]
+    ingress.metadata.annotations = {
+        INGRESS_CLASS: 'disabled',
+    }
+    ingress.metadata.name = 'test-app'
+    ingress.metadata.namespace = 'test-namespace'
+    return ingress
+
+
+@pytest.fixture
+def unidler_ingress():
+    host_rule = MagicMock()
+    host_rule.host = HOSTNAME
+    ingress = MagicMock()
+    ingress.spec.rules = [
+        host_rule,
+    ]
+    ingress.metadata.name = UNIDLER
+    ingress.metadata.namespace = 'default'
+    return ingress
+
+
+def test_remove_host_rule(unidler_ingress):
+    assert len(list(filter(
+        lambda rule: rule.host == HOSTNAME,
```

Comment:
Don't know if it's overkill, but you may put this lambda in a variable, e.g. `match_host` and reuse in both assertions.

Reply:
Yeah, I left a bunch of code to refactor in the tests - just wanted to get the tests working first.